lib/board.js

```diff
@@ -88,7 +91,14 @@ var Serial = {
         Serial.attempts[Serial.used.length]++;
 
         // Retry Serial connection
-        Serial.detect.call(this, callback);
+        if (Serial.attempts[Serial.used.length] > maxAttempts) {
+          this.fail("Board", "No connected device found");
```

Comment:
Here I would have logged the id of the board but I haven't because when initializing multiple boards without specifing ids they all end up having the same generated uid. Is this how it is supposed to be?